lib/reporting/authentication_report.rb

```diff
@@ -54,58 +54,72 @@ def progress?
       @progress
     end
 
-    # rubocop:disable Metrics/BlockLength
+    def as_tables
+      [
+        [
+          ['Report Timeframe', "#{time_range.begin} to #{time_range.end}"],
+          ['Report Generated', Date.today.to_s], # rubocop:disable Rails/Date
+          ['Issuer', issuers.join(', ')],
+          ['Total # of IAL1 Users', sp_redirect_initiated_all],
+        ],
+        [
+          ['Metric', 'Number of accounts', '% of total from start'],
+          [
+            'New Users Started IAL1 Verification',
+            email_confirmation,
+            format_as_percent(numerator: email_confirmation, denominator: email_confirmation),
+          ],
+          [
+            'New Users Completed IAL1 Password Setup',
+            two_fa_setup_visited,
+            format_as_percent(numerator: two_fa_setup_visited, denominator: email_confirmation),
+          ],
+          [
+            'New Users Completed IAL1 MFA',
+            user_fully_registered,
+            format_as_percent(numerator: user_fully_registered, denominator: email_confirmation),
+          ],
+          [
+            'New IAL1 Users Consented to Partner',
+            sp_redirect_initiated_new_users,
+            format_as_percent(
+              numerator: sp_redirect_initiated_new_users,
+              denominator: email_confirmation,
+            ),
+          ],
+          [
+            'AAL2 Authentication Requests from Partner',
+            oidc_auth_request,
+            format_as_percent(numerator: oidc_auth_request, denominator: oidc_auth_request),
+          ],
+          [
+            'AAL2 Authenticated Requests',
+            sp_redirect_initiated_after_oidc,
+            format_as_percent(
+              numerator: sp_redirect_initiated_after_oidc,
+              denominator: oidc_auth_request,
+            ),
+          ],
+        ],
+      ]
+    end
+
+    def as_tables_with_options
+      as_tables.each_with_index.map do |table, index|
+        title = (index == 0) ? 'Overview' : 'Authentication Funnel Metrics'
+        table.unshift({ title: })
+      end
```

Comment:
I would consider `.zip` here
```suggestion
      as_tables.zip(
        [
          { title: 'Overview' },
          { title: 'Authentication Funnel Metrics' },
        ],
      ).map do |table, options|
        [options, *table]
      end
```

Reply:
hm i personally find `zip` hard to reason about so i tend to avoid it. but, it does make it more extendable in the event we add more tables, and it's less clunky, so i will update it!

Reply:
ohhhhh yeah this i like a lot!!

Reply:
updated in https://github.com/18F/identity-idp/pull/9295/commits/688a701f41a6c6d6d6bfbee9b9f309ba66b44961